src/controllers/blocks/BlocksController.ts

```diff
@@ -263,9 +280,28 @@ export default class BlocksController extends AbstractController<BlocksService>
 			paraId: paraIdArg,
 		};
 
+		// Create a key for the cache that is a concatenation of the block hash and all the query params included in the request
+
+		const cacheKey =
+			hash.toString() +
+			Number(eventDocs) +
```

Comment:
Same with this.

Reply:
I agree with Tarik on this as well. You can also use `options.eventDocs`.